packages/spacecat-shared-data-access/test/unit/models/api-key.test.js

```diff
@@ -149,5 +149,42 @@ describe('ApiKey Model tests', () => {
     it('fails to update deletedAt with invalid date', () => {
       expect(() => apiKey.updateDeletedAt('invalid-date')).to.throw('Invalid deletedAt during update: invalid-date. Must be a valid ISO 8601 date string');
     });
+
+    it('fails to update expiresAt with invalid date', () => {
+      expect(() => apiKey.updateExpiresAt('invalid-date')).to.throw('Invalid expiresAt during update: invalid-date. Must be a valid ISO 8601 date string');
+    });
+
+    it('fails to update revokedAt with invalid date', () => {
+      expect(() => apiKey.updateRevokedAt('invalid-date')).to.throw('Invalid revokedAt during update: invalid-date. Must be a valid ISO 8601 date string');
+    });
+
+    it('returns true if deletedAt, revokedAt, and expiresAt are not set', () => {
+      apiKey.state.deletedAt = null;
+      apiKey.state.revokedAt = null;
+      apiKey.state.expiresAt = null;
+      expect(apiKey.isValid()).to.be.true;
+    });
+
+    it('returns false if deletedAt is after the current time', () => {
+      apiKey.updateDeletedAt(new Date(Date.now() + 10000).toISOString());
```

Comment:
Can concrete values please be used for these test cases so they are easier for a human to reason about?

ie. if a key expires in 2025 and it is 2024, it should still be valid.

Reply:
Done.